src/useCache.ts

```diff
@@ -1,17 +1,33 @@
-const cache: { [k: string]: { value: any; expiry: number } } = {}
-
 export const useCache = async (
+  KV: KVNamespace,
+  context: ExecutionContext,
   key: string,
   callback: Function,
   expirySeconds = 3600
 ) => {
-  const currentTime = new Date().valueOf() / 1000
-  const cached = cache[key]
-  if (cached && cached.expiry >= currentTime) return cached.value
-  cache[key] = { value: await callback(), expiry: currentTime + expirySeconds }
-  return cache[key].value
+  try {
+    const cached = await KV.get(key)
+    if (cached) return cached
+
+    const valueToCache = await callback()
+
+    if (valueToCache) {
```

Comment:
I think we do not need this `if` statement - we will also want to cache if the result of the callback is `false`, `undefined` or `null`, for example.

Reply:
in this case I've seen that the get returns a string 'undefined', which is not very helpful and I was thinking it's just better not to save null values, since they're anyway null(I know it would save some processing time since it would use the cached undefined value)